crates/workspace/src/workspace.rs

```diff
@@ -941,6 +941,25 @@ impl Workspace {
                     );
                 }
 
+                project::Event::LanguageServerShowDocument(request) => {
+                    if request.external {
+                        cx.open_url(request.uri.as_str());
+                        let request = request.clone();
+                        cx.background_executor()
+                            .spawn(async move {
+                                request.respond(Ok(())).await;
+                            })
```

Comment:
`open_url` doesn't return a status, so there's no way to tell if the link actually opened successfully. We assume it did and return that it did.